javascript/app-framework/src/components/modals/RenameModal/RenameModal.types.ts

```diff
@@ -0,0 +1,12 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { Resource } from '@datashaper/workflow'
+
+export interface RenameModalProps {
+	resource?: Resource
+	isOpen: boolean
+	onDismiss: () => void
+	onAccept: (newName: string | undefined) => void
```

Comment:
how can the new name be undefined?

Reply:
It's cancelled in this case

Reply:
As a client, you'd get a rejected promise

Reply:
I was thinking more along the lines of: if there is no value in the rename box, the modal Save button is disabled, so there should be no scenario where an empty name makes it through. Their only option if empty is to dismiss, calling onDismiss and rejecting the promise